python/ngen_cal/src/ngen/cal/meta.py

```diff
@@ -1,172 +1,49 @@
 import json
 import pandas as pd # type: ignore
 from pathlib import Path
+from tempfile import mkdtemp
 from typing import TYPE_CHECKING
 from .configuration import General, Model
 
 if TYPE_CHECKING:
     from pathlib import Path
     from pandas import DataFrame
 
-
-class CalibrationMeta:
+class JobMeta:
     """
-        Structure for holding calibration meta data
-
-        ###TODO can we hold enough `configuration` information to make it possible to update global config
+        Structure for holding model job meta data
     """
 
-    def __init__(self, model: Model, general: General):
-        """
+    def __init__(self, name: str, parent_workdir: Path, workdir: Path=None, log=False):
+        """Create a job meta data structure
 
+        Args:
+            name (str): name of the job, is used to construct log files
+            workdir (Path): working directory to stage the job under
+            log (bool, optional): Whether or not to create a log file for the job. Defaults to False.
         """
-        self._workdir = general.workdir
-        self._log_file = general.log_file
-        self._general = general
-        if(self._log_file is not None):
-            self._log_file = self._workdir/self._log_file
-        self._model = model #This is the Model Configuration object that knows how to operate on model configuration files
-        if self._general.strategy.target == 'max':
-            self._best_score = float('-inf')
-        else: #must be min or value, either way this works
-            self._best_score = float('inf')
-        self._best_params_iteration = '0' #String representation of interger iteration
-        self._bin = model.get_binary()
-        self._args = model.get_args()
-        
-        self._id = general.name #a unique identifier to prepend to log files
-        #FIXME another reason to refactor meta under catchment, logs per catchment???
-        if general.parameter_log_file is None:
-            self._param_log_file = self._workdir/"{}_best_params.txt".format(self._id)
-        else:
-            self._param_log_file = self._workdir/general.parameter_log_file
-        if general.objective_log_file is None:
-            self._objective_log_file = self._workdir/"{}_objective.txt".format(self._id)
+        if(workdir is None):
+            self._workdir = Path( mkdtemp(dir=parent_workdir, prefix=name+"_", suffix="_worker") ).resolve()
```

Comment:
Can we make this prefix include YYYYMMDDHHmm_ datetime? This would make sorting out multiple worker dir sets from different runs much easier.

Reply:
added in fbc54e97f8dfae336c8e29618a2c191045a3f970

Reply:
and fixed in 8d4ba5b786d929f3f39bdaa15c5987a6adbcab7e